process/block/metablock.go

```diff
@@ -1222,7 +1222,7 @@ func (mp *metaProcessor) CommitBlock(
 		"round", headerHandler.GetRound(),
 		"nonce", headerHandler.GetNonce(),
 		"hash", headerHash)
-
+	mp.wasBlockCommitted.SetValue(true)
```

Comment:
can you change so we save the first committed block and then only allow revert if there is a delta

Reply:
And this delta should be used only if we try to revert behind final, otherwise we should permit reverting as the first committed nonce could be greater with several nonces than the final nonce in other regular cases